RELEASES.md

```diff
@@ -0,0 +1,87 @@
+
+## Supported Versions
+
+The SOCI projct is still under rapid development. As such, official SOCI project support will be in the form of new Major and Minor versions. We may release Patch versions in an *ad hoc* manner if there is sufficient demand from the community.
+
+This policy will be changed once SOCI reaches v1.0.0.
+
+## Releases
+
+Releases are made through [GitHub Releases](https://github.com/awslabs/soci-snapshotter/releases). If there is demand, we will consider creating official linux packages for various distributions.
+
+
+### Major Version Releases (branch: `main`)
+
+Major versions are developed on the main branch. 
+
+When the time comes to release a new major version, a new `release/Major.0` branch will be created following the same process used for minor releases. 
+
+### Minor Version Releases (branch: `main`)
+
+Minor versions are developed on the main branch. 
+
+When the time comes to release a new minor version, a new `release/Major.Minor` branch will be created from the tip of main. Once the new branch has been created, a new `vMajor.Minor.0` tag will be created following the process used for patch releases.
+
+### Patch Version Releases (branch: `release/Major.Minor`)
+
+Patch versions are developed after the initial minor version tag in the `release/MAJOR.MINOR` branch to which they belong. Patch releases are used for security fixes and major or widespread bug fixes. Patch releases will not include new features. 
+
+When the time comes to release a new patch version, a new commit will be added to the `release/Major.Minor` branch to add any necessary release artifacts (e.g. a finalized copy of third party licenses, finalized change log, etc), and a new `vMajor.Minor.Patch` tag will be created from that commit. 
+
+Once the tag is created, a release will automatically be added to github with the same `vMajor.Minor.Patch` version scheme containing the release artifacts. For a full list of artifacts contained in the release, see [Release Artifacts](#release-artifacts).
+
+*__NOTE:__ release automation is currently aspirational. See https://github.com/awslabs/soci-snapshotter/issues/447*
+
+## API Stability
+
+Semantic Versioning expects minor versions to contain only backwards-compatible, new features. In order to avoid partial features from preventing new minor version releases, we do not guarantee that new features are stable when they are introduced. Instead, we maintain the following table indicating the stability of each new feature along with which SOCI version stabilized the feature.
+
+| Feature/API      | Stability | When it became stable         |
+|------------------|-----------|-------------------------------|
+| zTOC data format | Beta      | targetting before SOCI v1.0.0 |
+| zTOC API         | Unstable  | targetting v1.0.0             |
+| SOCI API         | Unstable  | targetting v1.0.0             |
+| soci CLI         | Unstable  |                               |
```

Comment:
I assume this refers to the 3 components mentioned in

"The SOCI CLI, snapshotter, and library all share the same version to make it easy to understand compatibility. Collectively, this will be referred to as the "SOCI version"."

If so I would suggest to:

1. zTOC API -> SOCI library: "zTOC API" is not mentioned anywhere except here. And it's mentioned in a Note that "but the library that's used to build ... zTOCs is part of the SOCI library and follows the SOCI version"
2. SOCI API -> SOCI snapshotter: I feel "SOCI API" and "SOCI library" refers to the same thing, e.g., the codebase/repo where we built SOCI. And we mentioned "SOCI snapshotter" is part of SOCI verion, but not mention it in the table.
3. soci CLI -> SOCI CLI: to be consistent.

This will ensure consistency with the "### SOCI CLI / snapshotter / library" section

Reply:
I'm okay with either with or without `snapshotter`. Instead, we should keep it consistent wherever we mention what SOCI version includes (e.g. in "### SOCI CLI / snapshotter / library" section and in the table).

I suggest we should have a list of components that have versions and their explanation, e.g.,:

```markdown
In SOCI, the following components are versioned:

1. zTOC API: what is zTOC API, or an example, or codepath.
2. SOCI CLI: ./cmd/soci.
3. SOCI API: what is SOCI API.
4. zTOC data format: maybe point to our zTOC flatbuf definition.

zTOC data format has its own version; the remaining components will be collectively referred to as the "SOCI version".
```

Then in the table, we can have the same list of components and track their stability.

Reply:
This is modeled on this section of the containerd releases doc: https://github.com/containerd/containerd/blob/main/RELEASES.md#public-api-stability

The purpose is for us to be able to develop new features on main. Our versioning says that minor version bumps will be backwards compatible, but this gives us an escape hatch for unfinished features. Collectively, those things are the library which has 1 version. 

I do think we should add sections after this table explaining what the APIs are, but I don't think we need to replace the word "library" in the rest of the doc with the list of APIs that make up that library.

Reply:
I added short descriptions of each of these components. I think there's still work to be done there, but I'm not entirely sure what information would be useful yet.